src/core/proof.ml

```diff
@@ -9,45 +9,58 @@ open Print
 (** Abstract representation of a goal. *)
 module Goal :
   sig
-    (** Representation of a goal. *)
-    type t
+    (** Representation of a proof type goal. *)
+    type goal_typ = (* TODO have to hide this *)
+      { goal_meta : meta  (* Metavariable needing instantiation.          *)
+      ; goal_hyps : Env.t (* Precomputed scope for a suitable term.       *)
+      ; goal_type : term  (* Precomputed type for a suitable term.        *) }
 
-    (** [of_meta m] create a goal from the metavariable [m]. *)
-    val of_meta : meta -> t
+    (** Representation of a general goal : type, unification *)
+    type t =
+      | GoalTyp of goal_typ (* The usual proof type goal. *)
```

Comment:
If Goal is never open then we could rename GoalTyp into Typ and GoalUnif into Unif, and use Goal.Typ and Goal.Unif instead of Goal.GoalTyp and Goal.GoalUnif.

Reply:
fixed in 67fcfa7